File: cses/trailingZerosLent.java
import java.math.BigInteger;
import java.util.*;

public class trailingZerosLent {

    static Scanner in;

    public static void main(String[] args) {
        
        in = new java.util.Scanner(System.in);

        int n = in.nextInt();
        
        BigInteger factorial = BigInteger.ONE;
        for (int i = 1; i<=n; i++) {
            factorial = factorial.multiply(BigInteger.valueOf(i));
        }

        //System.out.println(factorial);


        int count = 0;
        while (factorial.mod(BigInteger.TEN).equals(BigInteger.ZERO)) {
            count++;
            factorial = factorial.divide(BigInteger.TEN);
        }
            

        System.out.println(count);

    }
}

File: cses/trailingZerosLent.cpp
#include <iostream>
using namespace std;

int main() {

    int n;
    cin >> n;

    long long factorial = 1LL;
    for (int i = 1; i<=n; i++) {
        factorial *= i;
    }

    int count = 0;
    while (factorial%10 == 0) {
        count++;
        factorial /= 10;
    }

    cout << count << endl;

    return 0;
}